src/renderer/views/Detail/Detail.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { useState, useEffect, useCallback } from 'react';
+import { useSelector } from 'react-redux';
+import { useParams } from 'react-router-dom';
+import { Section, Label, Input } from '../../components/Setting/Setting';
+import { db } from '../../lib/app';
+import Button from '../../elements/Button/Button';
+import { RootState } from '../../store/reducers';
+import appStyles from '../../App.module.css';
+import * as LibraryActions from '../../store/actions/LibraryActions';
+import * as coverUtils from '../../../shared/lib/utils-cover';
+import Placeholder from '../../../images/assets/placeholder.png';
+import { getLoweredMeta, parseDuration } from '../../lib/utils';
+
+import styles from './Detail.module.css';
+
+const Detail: React.FC = () => {
+  const { trackId } = useParams<{ trackId: string }>();
+  const [track, setTrack] = useState(undefined);
+  const [coverSrc, setCoverSrc] = useState(undefined);
+  const [data, setData] = useState({
+    title: '',
+    artist: '',
+    album: '',
+    genre: '',
+    duration: '',
+  });
+
+  const getCover = useCallback(async (path: string) => {
+    const data = await coverUtils.fetchCover(path);
+    if (data !== null) {
+      setCoverSrc(data);
+    }
+  }, []);
+
+  const updateSong = useCallback(
+    (data) => {
+      const song = { ...track };
+      song.title = data.title;
+      song.artist[0] = data.artist;
+      song.album = data.album;
+      song.genre[0] = data.genre;
+      song.loweredMetas = getLoweredMeta(song);
+      return song;
+    },
+    [track]
+  );
+
+  const handleSubmit = useCallback(
+    async (e) => {
+      e.preventDefault();
+      const song = updateSong(data);
+      await LibraryActions.updateTrack(song);
+      history.back();
+    },
+    [data, updateSong]
+  );
+
+  const handleChange = useCallback(
+    (e) => {
+      e.preventDefault();
+      setData({
+        ...data,
+        [e.target.name]: e.target.value,
+      });
+    },
+    [data]
+  );
+
+  const handleCancel = useCallback((e) => {
+    e.preventDefault();
+    history.back();
+  }, []);
+
+  useEffect(() => {
+    db.Track.findOne({ _id: trackId }, (err, song) => {
+      if (err !== null) return console.log(err);
+      getCover(song.path);
+
+      setTrack(song);
+      setData({
+        title: song.title || '',
+        artist: song.artist[0] || '',
+        album: song.album || '',
+        genre: song.genre[0] || '',
+        duration: parseDuration(song.duration) || '',
+      });
+    });
```

Comment:
I just realized we're only updating the DB but not the file, I'll have to do some research on how we can update the files as well.

Reply:
you're right 😆. I'll solve this question on the next commit.

Regards.  😊

Reply:
solved with  [9a40bece](https://github.com/martpie/museeks/pull/604/commits/9a40bece78c26441129fe475f898657b50cb10e4)